tests/unit/amo/components/TestAddonReviewList.js

```diff
@@ -46,12 +46,21 @@ describe(__filename, () => {
     store = dispatchClientMetadata({ clientApp, lang }).store;
   });
 
-  const getProps = ({ ...customProps }) => {
+  const getProps = ({ location, params, router, ...customProps } = {}) => {
     return {
       i18n: fakeI18n(),
-      location: { query: {} },
-      params: {
-        addonSlug: fakeAddon.slug,
+      // The `withRouter()` HOC uses the `router` to pass `params` and
+      // `location` to its children.
+      router: {
```

Comment:
It's a shame that we have to mock the whole router object but I don't have a better solution. I thought it would be possible to start the test with:

```js
import { push } from 'react-router-redux';
...
store.dispatch(push(`/en-US/firefox/addon/${fakeAddon.slug}/reviews/?page=2`))
```

...but I fell down a rabbit hole trying to get that to work. Oh well.

Reply:
Yeah not having to mock it would be great. Maybe at least a comment or an issue encouraging us to try to be better in the future 😉